src/AcceptanceTests/When_session_is_accessed_in_hosted_service_ctor.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace AcceptanceTests
+{
+    using Microsoft.Extensions.Hosting;
+    using NUnit.Framework;
+    using NServiceBus;
+    using Microsoft.Extensions.DependencyInjection;
+    using System.Threading.Tasks;
+    using System.Threading;
+    using System;
+
+    [TestFixture]
+    public class When_session_is_accessed_in_hosted_service_ctor
+    {
+        [Test]
+        public void Should_throw()
+        {
+            var ex = Assert.ThrowsAsync<InvalidOperationException>(async () =>
+            {
+                var host = Host.CreateDefaultBuilder()
+                .UseNServiceBus(hostBuilderContext =>
+                {
+                  var endpointConfiguration = new EndpointConfiguration("MyEndpoint");
+                  endpointConfiguration.SendOnly();
+                  endpointConfiguration.UseTransport<LearningTransport>();
+                  return endpointConfiguration;
+                })
+                .ConfigureServices((ctx, serviceProvider) => serviceProvider.AddHostedService<HostedServiceThatAccessSessionInCtor>()).Build();
+
+                await host.StartAsync();
```

Comment:
This test doesn't stop the host, should that be added?

Reply:
since this is expected to throw the stop would never be called?